### PR TITLE
Upgrade PHPWord

### PR DESCRIPTION
**Pull request by seamuslee001**

### CODE REVIEW

File: CRM/Core/Menu.php [hunk 1]
```diff
@@ -112,8 +112,10 @@ public static function &xmlItems($fetchFromXML = FALSE) {
    * @throws Exception
    */
   public static function read($name, &$menu) {
+    libxml_disable_entity_loader(FALSE);
     $xml = simplexml_load_file($name);
     self::readXML($xml, $menu);
+    libxml_disable_entity_loader(TRUE);
```
Comment (totten) [inline CRM/Core/Menu.php]: The general pattern for doing this safely as well-behaved member of a complex system is:

```php
$backupValueThatIDontMakeAnyAssumptionsAbout = readTheGlobalValue():
setTheGlobalValue($theValueThatILike);
try {
   doTheMainThing();
}
finally {
  setTheGlobalValue($backupValueThatIDontMakeAnyAssumptionsAbout);
}
```

On a case-by-cases basis, there might be a prettier way (e.g. maybe depending on the contract of the specific global-thing or  maybe using `CRM_Utils_AutoClean::swap()` or maybe using `CRM_Utils_GlobalStack`).

**Comment by seamuslee001** [comment]
@totten got it to pass :-)

**Comment by eileenmcnaughton** [comment]
@seamuslee001 are there remaining reasons not to merge now?

**Comment by seamuslee001** [comment]
nothing on my side but would appreciate Tim's opinion as well

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by eileenmcnaughton** [comment]
We should actually go to the latest version I think
https://github.com/PHPOffice/PHPWord/releases

My experience with phpword is that we have had more than one case where the fix to some reported issue is to upgrade - as long as it's compatible with our min php version I think keeping it current solves more problems than it's likely to risk causing

**Comment by colemanw** [comment]
Looks like the latest is v0.16.

@seamuslee001 will that newer version still require your entity_loader patch?

**Comment by seamuslee001** [comment]
@colemanw yes it would still need those patches

**Comment by eileenmcnaughton** [comment]
So I'm happy to merge this as a 'technical' fix to the problem of 'how do we upgrade phpword - although getting the latest bug fixes would be better

**Pull request merged**